tests/network/test_network.py

```diff
@@ -523,3 +523,70 @@ async def test_network_should_not_block_queue_shared_workers_with_data_url(
             "timestamp": ANY_TIMESTAMP
         }
     })
+
+
+@pytest.mark.asyncio
+async def test_network_preflight(websocket, context_id, html, url_example,
+                                 read_messages):
+    """ https://github.com/GoogleChromeLabs/chromium-bidi/issues/3570 """
+    url = html("hello world!",
+               same_origin=False,
+               headers={
+                   "Access-Control-Allow-Origin": "*",
+                   "Access-Control-Allow-Headers": "x-ping"
+               })
+
+    await goto_url(websocket, context_id, url)
+
+    await subscribe(websocket, ["network.beforeRequestSent"])
+
+    await send_JSON_command(
+        websocket, {
+            "method": "script.evaluate",
+            "params": {
+                "expression": f"""
+                    fetch('{url_example}', {{
+                        method: 'POST',
+                        headers: {{'x-ping': 'pong'}},
+                    }})
+                """,
+                "target": {
+                    "context": context_id
+                },
+                "awaitPromise": False,
+            }
+        })
+
+    [_, preflight_request, post_request] = await read_messages(3, sort=False)
+
+    assert preflight_request == AnyExtending({
```

Comment:
this probably will be flaky just like in the protocol tests

Reply:
yup, fixed